operator/cmd/mesh/manifest-versions.go

```diff
@@ -79,6 +81,38 @@ func manifestVersions(args *rootArgs, mvArgs *manifestVersionsArgs, l *Logger) e
 	return nil
 }
 
+// CheckK8sVersion checks if the current client and server version is supported
+func CheckK8sVersion(opts *kubectlcmd.Options, l *Logger) error {
+	cK8sVer, sK8sVer, err := manifest.Version(opts)
+	if err != nil {
+		return fmt.Errorf("failed to read versions from client and cluster: %v", err)
+	}
+	compatibleMap, err := getVersionCompatibleMap("", binversion.OperatorBinaryGoVersion, l)
+	if err != nil {
+		return err
+	}
+	return checkK8sVersion(compatibleMap, cK8sVer, sK8sVer)
+}
+
+// checkK8sVersion checks if the client and server versions are supported in CompatibilityMapping
+func checkK8sVersion(vmap *version.CompatibilityMapping, cVer, sVer *goversion.Version) error {
+	if cVer == nil {
+		return fmt.Errorf("failed to get k8s client version, please check if kubectl is correctly installed")
+	}
+	if sVer == nil {
+		return fmt.Errorf("failed to get k8s server version, please check if the k8s context is configured")
+	}
+	if !vmap.K8sClientVersionRange.Check(cVer) {
```

Comment:
When would the client version check fail? This is built into to the istioctl binary itself.

Reply:
Istioctl is still calling kubectl, that’s why some users are reporting issues when they have old version kubectl installed.

Reply:
for example, https://github.com/istio/istio/issues/20089

Reply:
the reason we haven't switched to k8s go-client yet: we experienced some slowness issue when using go-client in controller mode, most likely caused by the server-side throttle rules from api server.